### PR TITLE
[stable/ark] upgrade 0.9 to 0.10 + restic support

### PR DESCRIPTION
**Pull request by agolomoodysaada**
Signed-off-by: moody <moody.saada@agolo.com>

#### Special notes for your reviewer:

Some old flags are no longer supported in 0.10 given the breaking changes.
For example: https://github.com/heptio/ark/pull/757/files#diff-885cfbae1bf8bdbfbc9fcd15b8eda84eL36

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/ark/templates/deployment.yaml [hunk 1]
```diff
@@ -35,6 +35,23 @@ spec:
...
+            - --metrics-address={{ . }}
+            {{- end }}
+            {{- with .backupSyncPeriod }}
+            - --backup-sync-period={{ . }}
+            {{- end }}
+            {{- with .resticTimeout }}
+            - --restic-timeout={{ . }}
+            {{- end }}
+            {{- with .restoreOnlyMode }}
+            - --restore-only={{ . }}
```
Comment (spingel) [inline stable/ark/templates/deployment.yaml]: This won't work if restoreOnlyMode=false is specified. The --restore-only parameter does not accept a value but defaults to true if specified. I would suggest changing this to the following:

`            {{- if .restoreOnlyMode }}`
`            - --restore-only`
`            {{- end }}`

File: stable/ark/values.yaml [hunk 2]
```diff
@@ -38,11 +38,13 @@ configuration:
...
     #  kmsKeyId:
+    #  resourceGroup:
+    #  storageAccount:
 
   backupSyncPeriod: 60m
-  gcSyncPeriod: 60m
-  scheduleSyncPeriod: 1m
-  resourcePriorities: []
+  metricsAddress: ":8085"
+  resticTimeout: 1m
```
Comment (spingel) [inline stable/ark/values.yaml]: This seems very low. `ark server --help` shows a default of `1h`.

**Review by spingel — COMMENTED** [review]
This change looks great!

File: stable/ark/templates/deployment.yaml (hunk 1 above)
Comment (agolomoodysaada) [reply stable/ark/templates/deployment.yaml]: changed

File: stable/ark/values.yaml (hunk 2 above)
Comment (agolomoodysaada) [reply stable/ark/values.yaml]: changed

File: stable/ark/templates/backupstoragelocations.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: backupstoragelocations.ark.heptio.com
+  labels:
+    chart: {{ template "ark.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    app: {{ template "ark.name" . }}
+  annotations:
+    "helm.sh/hook": crd-install
```
Comment (cpanato) [inline stable/ark/templates/backupstoragelocations.yaml:11]: please add ` "helm.sh/hook-delete-policy": "before-hook-creation"`

Comment (agolomoodysaada) [reply stable/ark/templates/backupstoragelocations.yaml:11]: updated

**Comment by cpanato** [comment]
@agolomoodysaada I saw there are other CRDs, we might need to apply the same in the others

**Comment by agolomoodysaada** [comment]
@cpanato , thanks for the feedback. What do you mean "apply the same..."? the same labels? the same configuration?

**Comment by cpanato** [comment]
@agolomoodysaada sorry for the delay

I saw other CRDs in this chart, we might want to add `"helm.sh/hook-delete-policy": "before-hook-creation"` for all of them

**Comment by agolomoodysaada** [comment]
@cpanato , test is still failing even with the added annotations. Is this a bug in Helm?

**Comment by cpanato** [comment]
@agolomoodysaada will check in the cluster

**Pull request merged**